engine/execution/ingestion/core.go

```diff
@@ -43,15 +43,14 @@ type Core struct {
 	// will be pushed to the blockExecutors channel, and the worker will execute the block.
 	// during startup, the throttle will limit the number of blocks to be added to the processables channel.
 	// once caught up, the throttle will allow all the remaining blocks to be added to the processables channel.
-	processables   chan flow.Identifier         // block IDs that are received and waiting to be processed
+	processables   chan BlockIDHeight           // block IDs that are received and waiting to be processed
 	throttle       Throttle                     // to throttle the blocks to be added to processables during startup and catchup
 	blockQueue     *block_queue.BlockQueue      // blocks are waiting for the data to be fetched
 	blockExecutors chan *entity.ExecutableBlock // blocks that are ready to be executed
 	stopControl    *stop.StopControl            // decide whether to execute a block or not and when to stop the execution
 
 	// data storage
 	execState   state.ExecutionState
-	headers     storage.Headers
```

Comment:
Since processables are a channel of `BlockIDHeight`, which contains block height, we no longer need `headers` to query block height by ID